src/Libraries/Microsoft.Extensions.AI/ChatCompletion/OpenTelemetryChatClient.cs

```diff
@@ -117,35 +135,37 @@ public override async Task<ChatCompletion> CompleteAsync(IList<ChatMessage> chat
         }
         finally
         {
-            SetCompletionResponse(activity, requestModelId, response, error, stopwatch);
+            TraceCompletion(activity, requestModelId, completion, error, stopwatch);
         }
-
-        return response;
     }
 
     /// <inheritdoc/>
     public override async IAsyncEnumerable<StreamingChatCompletionUpdate> CompleteStreamingAsync(
         IList<ChatMessage> chatMessages, ChatOptions? options = null, [EnumeratorCancellation] CancellationToken cancellationToken = default)
     {
+        _ = Throw.IfNull(chatMessages);
         _jsonSerializerOptions.MakeReadOnly();
 
-        using Activity? activity = StartActivity(chatMessages, options);
+        using Activity? activity = CreateAndConfigureActivity(options);
         Stopwatch? stopwatch = _operationDurationHistogram.Enabled ? Stopwatch.StartNew() : null;
         string? requestModelId = options?.ModelId ?? _modelId;
 
-        IAsyncEnumerable<StreamingChatCompletionUpdate> response;
+        LogChatMessages(chatMessages);
+
+        IAsyncEnumerable<StreamingChatCompletionUpdate> updates;
         try
         {
-            response = base.CompleteStreamingAsync(chatMessages, options, cancellationToken);
+            updates = base.CompleteStreamingAsync(chatMessages, options, cancellationToken);
         }
         catch (Exception ex)
         {
-            SetCompletionResponse(activity, requestModelId, null, ex, stopwatch);
+            TraceCompletion(activity, requestModelId, completion: null, ex, stopwatch);
             throw;
         }
 
-        var responseEnumerator = response.ConfigureAwait(false).GetAsyncEnumerator();
-        List<StreamingChatCompletionUpdate>? streamedContents = activity is not null ? [] : null;
+        var responseEnumerator = updates.ConfigureAwait(false).GetAsyncEnumerator();
```

Comment:
I've never seen `ConfigureAwait` being applied to an IAE. Does the remove the need for doing `ConfigureAwait` on individual `MoveNextAsync()` calls or is it something else?

Reply:
Right, it effectively adds the ConfigureAwait onto each await performed implicitly as part of the foreach.